File: python solutions/twoSum.py
##nums is an array of numbers
##target is the target numbers

def twoSums(nums, target):
  ##[2, 4, 5, 8]
  ##create a dictionary to store already checked values in the nums list
  seen = {}
  
  ##creating a for...in loop to loop through each value in the array
  for i, num in enumerate(nums):
      ##get the complement number e.g for a target of 9, the compliment is 9-4 = 5
      compliment = target - nums[i]
      
      if  compliment in seen:
          return [seen[compliment], i]
      seen[num] = i
  print(seen)
  return []
    
result = twoSums([2, 7, 5, 8], 9)

print(result)

File: TwoSum.js
function TwoSumProblem(nums, target){
    //create a map to store already checked values in the array and its index
    //sample { 4: 0, 2: 1 } for an array [4, 2]
    let map = new Map();

    for(let i = 0; i < nums.length; i++){
        //get the compliment number e.g for a target of 9, the compliment is 9-4 = 5
        const compliment = target - nums[i];

        //check if the compliment exists in the map
        if(map.has(compliment)){
            return [map.get(compliment), i]
        }

        map.set(nums[i], i)
    }
    return [];
}